hub/core/chunk_engine/read.py

```diff
@@ -4,12 +4,21 @@
 
 from .chunker import join_chunks
 
+from hub import constants
 from hub.util.keys import get_meta_key, get_index_map_key
 
 from hub.core.typing import StorageProvider
 from typing import Callable, List, Union
 
 
+def read_tensor_meta(key: str, storage: StorageProvider):
+    return pickle.loads(storage[get_meta_key(key)])
+
+
+def read_dataset_meta(storage: StorageProvider):
```

Comment:
i don't think the "read_dataset_meta" function should be in chunk_engine. this seems out of scope.

Reply:
Where would you like it? I thought next to `read_array` was relevant, is there a better place?

Reply:
something like `hub.core.structure_engine.meta.py` would be appropriate. we can rename `structure_engine` later but for now we should make note that it is separate.

Reply:
and technically this falls under the `structure_engine`. all it does is reads/writes dataset metas and translates indexing into a new form

Reply:
See below thread